src/util/domain_info_utils.c

```diff
@@ -521,24 +521,24 @@ sss_write_domain_mappings(struct sss_domain_info *domain)
         goto done;
     }
 
-    ret = rename(tmp_file, mapping_file);
+    ret = chmod(tmp_file, 0644);
```

Comment:
TOCTOU still exist unless `fchmod` is used and with it the order is irrelevant.

Reply:
Sorry, I still fail to understand value of the patch.

Granting access to the file (`go+r`) after file is fully prepared (renamed) makes most sense from my point of view (as it is in the original code).

Reply:
Why does it make more sense? Is there any guide/best practices that recommends doing it that way?

Reply:
Let me put it other way around: what could be a reason to grant (extend) access to a temporary file?

Reply:
Make sure that when the file is moved to its final destination it contains the correct permissions.

Reply:
> Make sure that when the file is moved to its final destination it contains the correct permissions.

Since this action loosens (not tightens) access rights, I don't see any benefit from security point of view.

Reply:
Change dropped.